scriptworker/github.py

```diff
@@ -132,21 +127,43 @@ def definition(self):
                      "repositories, assume True")
             return True
 
-        url = '/'.join([repo.rstrip('/'), 'branch_commits', revision])
-        from scriptworker.task import get_decision_task_id
-        cache_key = '{}-{}'.format(get_decision_task_id(context.task), url)
-        if cache_key in _branch_commits_cache:
-            html_text = _branch_commits_cache[cache_key]
-        else:
-            html_data = await retry_request(context, url)
-            html_text = html_data.strip()
-            _branch_commits_cache[cache_key] = html_text
+        # Revision may be a tag name. `branch_commits` doesn't work on tags
+        if not _is_git_full_hash(revision):
+            revision = await self.get_tag_hash(tag_name=revision)
+
+        html_text = await _memoized_fetch_github_branch_commits_data(
+            context, self._github_repository.html_url, revision
+        )
+
         # https://github.com/{repo_owner}/{repo_name}/branch_commits/{revision} just returns some \n
         # when the commit hasn't landed on the origin repo. Otherwise, some HTML data is returned - it
         # represents the branches on which the given revision is present.
         return html_text != ''
 
 
+# XXX Please use `_memoized_fetch_github_branch_commits_data()` instead. Defining 2 functions won't
+# be necessary anymore if memoize_ttl() is available as a function decorator.
+# See https://github.com/michalc/aiomemoizettl/issues/2
+async def _fetch_github_branch_commits_data(context, repo_html_url, revision):
+    url = '/'.join((repo_html_url.rstrip('/'), 'branch_commits', revision))
+    log.info('Cache does not exist for URL "{}" (in this context), fetching it...'.format(url))
```

Comment:
Cool, so we can look for duplicates of this line if we're debugging the memoization.